### ISSUE TITLE
x265 fails to install

### ISSUE THREAD

**Issue by AlexanderOMara** [Linuxbrew/homebrew-core#4717]
**Checklist:**
- [x] Confirmed this is a problem with `brew install`ing one, specific formula and not every time you run `brew`?
- [x] Cannot replicate on a macOS system.
- [x] Ran `brew update` and retried your prior step?
- [x] Ran `brew doctor`, fixed all issues and retried your prior step?
- [x] Ran `brew gist-logs x265`: https://gist.github.com/anonymous/d7502f8dd124c67ee59a676bf20f24ef

To help us debug your issue please explain:
- What you were trying to do (and why)
  - Trying to upgrade x265
- What happened (include command output)
  - Failed to install.
- What you expected to happen
  - A successful install.
- Step-by-step reproduction instructions (by running `brew install` commands)
  - `brew install x265`

Console output:
```
$ brew upgrade x265
==> Upgrading 1 outdated package, with result:
x265 2.5_1
==> Upgrading x265
==> Downloading https://bitbucket.org/multicoreware/x265/downloads/x265_2.5.tar.gz
Already downloaded: /home/main/.cache/Homebrew/x265-2.5.tar.gz
==> cmake /tmp/x265-20171017-43472-1bposcu/x265_2.5/source -DCMAKE_C_FLAGS_RELEASE=-DNDEBUG -DCMAKE_CXX_FLAGS_RELEASE=-DNDEBUG -DCMAKE_INSTALL_PREFIX=/home/linuxbrew/.linuxbrew/Cellar/x265/2.5_1 -DCMAKE_BUILD_TYPE=Release -DCMAKE_FIND_FRAMEWORK=LAST -DCMAKE_VERBOSE_MAKEFILE=ON -Wno-dev -DHIGH_BIT_DEPTH=ON -DEXPORT_C_API=OFF -DENABLE_SHARED=OFF -DENABLE_CLI=OFF
==> make
==> cmake /tmp/x265-20171017-43472-1bposcu/x265_2.5/source -DMAIN12=ON -DCMAKE_C_FLAGS_RELEASE=-DNDEBUG -DCMAKE_CXX_FLAGS_RELEASE=-DNDEBUG -DCMAKE_INSTALL_PREFIX=/home/linuxbrew/.linuxbrew/Cellar/x265/2.5_1 -DCMAKE_BUILD_TYPE=Release -DCMAKE_FIND_FRAMEWORK=LAST -DCMAKE_VERBOSE_MAKEFILE=ON -Wno-dev -DHIGH_BIT_DEPTH=ON -DEXPORT_C_API=OFF -DENABLE_SHARED=OFF -DENABLE_CLI=OFF
==> make
==> cmake /tmp/x265-20171017-43472-1bposcu/x265_2.5/source -DCMAKE_C_FLAGS_RELEASE=-DNDEBUG -DCMAKE_CXX_FLAGS_RELEASE=-DNDEBUG -DCMAKE_INSTALL_PREFIX=/home/linuxbrew/.linuxbrew/Cellar/x265/2.5_1 -DCMAKE_BUILD_TYPE=Release -DCMAKE_FIND_FRAMEWORK=LAST -DCMAKE_VERBOSE_MAKEFILE=ON -Wno-dev -DLINKED_10BIT=ON -DLINKED_12BIT=ON -DEXTRA_LINK_FLAGS=-L. -DEXTRA_LIB=x265_main10.a;x265_main12.a
==> make
==> libtool -static -o libx265.a libx265_main.a libx265_main10.a libx265_main12.a
Last 15 lines from /home/main/.cache/Homebrew/Logs/x265/07.libtool:
2017-10-17 17:37:45 -0400

libtool
-static
-o
libx265.a
libx265_main.a
libx265_main10.a
libx265_main12.a


READ THIS: https://github.com/Linuxbrew/brew/blob/master/docs/Troubleshooting.md#troubleshooting
Please do not report this issue to Homebrew/brew or Homebrew/core, which support macOS only.

These open issues may also help:
x265: Build a bottle for Linuxbrew https://github.com/Linuxbrew/homebrew-core/pull/4648
```

**Comment by AlexanderOMara** [comment]
I ran `brew install --verbose --debug x265`, I believe the issue is related to libtool:

```
==> libtool -static -o libx265.a libx265_main.a libx265_main10.a libx265_main12.a
Usage: /usr/bin/libtool [OPTION]... [MODE-ARG]...
Try 'libtool --help' for more information.
libtool:   error: unrecognised option: '-static'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/build.rb (Formulary::FormulaLoader): loading /home/linuxbrew/.linuxbrew/Homebrew/Library/Taps/linuxbrew/homebrew-xorg/xorg.rb
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/debrew.rb:11:in `raise'
BuildError: Failed executing: libtool -static -o libx265.a libx265_main.a libx265_main10.a libx265_main12.a
1. raise
2. ignore
3. backtrace
4. irb
5. shell
Choose an action: 5
When you exit this shell, you will return to the menu.
To run a command as administrator (user "root"), use "sudo <command>".
See "man sudo_root" for details.

USER@SERVER:/tmp/x265-20171017-56371-9e9hfv/x265_2.5/8bit$ libtool -static -o libx265.a libx265_main.a libx265_main10.a libx265_main12.a
Usage: /usr/bin/libtool [OPTION]... [MODE-ARG]...
Try 'libtool --help' for more information.
libtool:   error: unrecognised option: '-static'
USER@SERVER:/tmp/x265-20171017-56371-9e9hfv/x265_2.5/8bit$ which libtool
/usr/bin/libtool
USER@SERVER:/tmp/x265-20171017-56371-9e9hfv/x265_2.5/8bit$
```
After playing around in the shell, I think the issue is related to this change, which appears dependent on a system libtool that supports a `-static` option: https://github.com/Linuxbrew/homebrew-core/commit/3d8f4662b51f088d8288e964c265b241309da7df

The libtool is the system one (installed by `sudo apt-get install libtool-bin`, and not installed by default on Ubuntu), and `-static` does not appear to be an option. I tried using the Homebrew libtool instead, but -static also does not appear to be an option there either.

**Comment by AlexanderOMara** [comment]
Oh, I see. The issue is that those system commands expect the Apple libtool, which has a `-static` and `-o` option. It seems those changes made the formula incompatible with Linux...

**Comment by AlexanderOMara** [comment]
IDK if this is a perfect solution, or how to make it part of the formula, but I've come up with a patch that can be applied to get it to install:

Edit: `/home/linuxbrew/.linuxbrew/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/x265.rb`

Replace:
```ruby
      system "libtool", "-static", "-o", "libx265.a", "libx265_main.a",
                        "libx265_main10.a", "libx265_main12.a"
```

With:
```ruby
      #system "libtool", "-static", "-o", "libx265.a", "libx265_main.a",
      #                  "libx265_main10.a", "libx265_main12.a"
      system "ar", "cr", "libx265.a", "libx265_main.a", "libx265_main10.a", "libx265_main12.a"
      system "ranlib", "libx265.a"
```

**Comment by iMichka** [comment]
Hi. Thanks for reporting this. I saw that it was broken 5 days ago, see #4648, and had no time to work on this (and will probably not have any time in the next days.

Would you be willing to open a pull request for this? You will need to create a if/else block using `if OS.mac?` as condition, because we want to keep the mac version around (this help us when later merges are done).

**Comment by AlexanderOMara** [comment]
I've created a pull request:

https://github.com/Linuxbrew/homebrew-core/pull/4731

**Comment by iMichka** [comment]
Fixed with #4731
